hack/golangci-lint

```diff
@@ -0,0 +1,22 @@
+#! /usr/bin/env bash
+
+readonly PROGNAME="golangci-lint"
```

Comment:
prefer if the project installs tooling on behalf of the user as shown [here](https://github.com/envoyproxy/gateway/pull/84)
rather than using the `hack` approach

Reply:
If we use `installs tooling` in Makefile instead of using `hack` approach, I think we need to wait until #84 has been merged.

On second thought, to run golangci-lint and yamllint, we simply just use `docker run` once, I don't think we need any installation steps to set up some tools first( not `go install` related). Maybe we should keep the `hack` approach in these situations, WDYT?